resources/bucket_list.py

```diff
@@ -0,0 +1,245 @@
+                else:
+                    url_next = 'Null'
+
+                if bucket_list.has_next:
+                    url_prev = request.url + '?page='+ str(page - 1) + '&limit=' + str(per_page)
+                else:
+                    url_prev = 'Null'
+
+                return {'meta':{'next_page':url_next,
+                                'previous_page':url_prev,
+                                'total_pages': bucket_list.pages},
+                        'bucketlist':marshal(bucket_list.items, bucket_list_field)}, 200
+
+            else:
+                return {'message': 'no bucket_list available'}, 200
+
+    def post(self):
+        """ Adds a bucketlist """
+
+        args = self.reqparse.parse_args()
+        name = args['name']
+
+        bucket_list = BucketList.query.filter_by(name=name).first()
+
+        if bucket_list:
+            return {'message':'bucketlist {0} already exists'.format(name)}, 202
+
+        new_bucket_list = BucketList(name=name, created_by=g.user.id)
+        db.session.add(new_bucket_list)
+        db.session.commit()
+
+        return marshal(new_bucket_list, bucket_list_field), 201
+
+    def put(self, id):
+        """ updates the bucketlist item of a specific id"""
+
+        args = self.reqparse.parse_args()
+        name = args['name']
+
+        bucket_list = BucketList.query.filter_by(id=id, created_by=g.user.id).first()
+
+        if bucket_list is None:
+            return {'error':'bucketlist does not exists'}, 404
+
+        bucket_list.name = name
+        db.session.commit()
+
+        return marshal(bucket_list, bucket_list_field), 200
+
+    def delete(self, id):
+        """ deletes bucketlist item of a particular id """
+
+        bucket_list = BucketList.query.filter_by(id=id, created_by=g.user.id).first()
+
+        if bucket_list:
+            db.session.delete(bucket_list)
+            db.session.commit()
+
+            return {'message':'bucket list {0} deleted successfully'.format(bucket_list.name)}, 200
+
+        return {'message':'bucket list does not exist'}, 404
+
+class BucketListItemAPI(Resource):
+    """ API endpoint for the bucketlist item"""
+
+    decorators = [auth.login_required]
+
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        super(BucketListItemAPI, self).__init__()
+
+    def post(self, bucketlist_id):
+        """ creates a new bucketlist item"""
+        self.reqparse.add_argument('item_name', type=str, required=True,
+                                   help='No item name provided',
+                                   location='json')
+        args = self.reqparse.parse_args()
+        name = args['item_name']
+        bucket_list = BucketList.query.filter_by(id=bucketlist_id).first()
+
+        if bucket_list:
+            bucket_list_item_last = BucketListItem.query.filter_by(bucketlist_id=bucketlist_id)\
+                                   .order_by(db.desc(BucketListItem.bucketlist_item_id))
+
```

Comment:
So consider using  parentheses instead of forward slash. This makes the code neater and readable.